File: src/_03_BehavioralPattern/_03_06_Memento/GameSave.js
export default class GameSave {
  /** @type { number } */
  _blueTeamScore;

  /** @type { number } */
  _redTeamScore;

  constructor(blueTeamScore, redTeamScore) {
    this._blueTeamScore = blueTeamScore;
    this._redTeamScore = redTeamScore;
  }

  /** @returns { number } */
  getBlueTeamScore() {
    return this._blueTeamScore;
  }

  /** @returns { number } */
  getRedTeamScore() {
    return this._redTeamScore;
  }
}

File: src/_03_BehavioralPattern/_03_06_Memento/java/after/GameSave.java
package _03_BehavioralPattern._03_06_Memento.java.after;

public class GameSave {
  
  private int blueTeamScore;

  private int redTeamScore;

  public GameSave(int blueTeamScore, int redTeamScore) {
    this.blueTeamScore = blueTeamScore;
    this.redTeamScore = redTeamScore;
  }

  public int getBlueTeamScore() {
    return this.blueTeamScore;
  }
  public int getRedTeamScore() {
    return this.redTeamScore;
  }
  
}
